app/com/linkedin/drelephant/spark/heuristics/StagesHeuristic.scala

```diff
@@ -196,15 +234,48 @@ object StagesHeuristic {
       val averageExecutorRuntime = stageData.executorRunTime / executorInstances
       (averageExecutorRuntime, stageRuntimeMillisSeverityThresholds.severityOf(averageExecutorRuntime))
     }
+
+    /**
+      * returns the total JVM GC Time and total executor CPU Time across all stages
+      * @param stageDatas
+      * @return
+      */
+    private def getTimeValues(stageDatas: Seq[StageData]): (Long, Long) = {
+      var jvmGcTimeTotal: Long = 0
+      var executorCpuTime: Long = 0
+      var taskMetricsDummy: TaskMetricsImpl = new Some(new TaskMetricsImpl(
+        executorDeserializeTime = 0,
+        executorRunTime = 0,
+        executorCpuTime = 0,
+        resultSize = 0,
+        jvmGcTime = 0,
+        resultSerializationTime = 0,
+        memoryBytesSpilled = 0,
+        diskBytesSpilled = 0,
+        inputMetrics = None,
+        outputMetrics = None,
+        shuffleReadMetrics = None,
+        shuffleWriteMetrics = None)).get
+      //ignoring the exception as there are cases when there is no task data, in such cases, 0 is taken as the default value
+      Exception.ignoring(classOf[NoSuchElementException]) {
+        stageDatas.foreach((stageData: StageData) => {
+          stageData.tasks.get.values.foreach((taskData: TaskData) => {
+            jvmGcTimeTotal += taskData.taskMetrics.getOrElse(taskMetricsDummy).jvmGcTime
```

Comment:
I don't think it's correct to add GC time for each task to get a total GC time. JVM GC time is a global entity if multiple tasks are running in parallel, they will have the same GC time, and adding them up will be overcounting.